homeassistant/components/notify/__init__.py

```diff
@@ -116,7 +116,6 @@ class BaseNotificationService:
 
     # While not purely typed, it makes typehinting more useful for us
     # and removes the need for constant None checks or asserts.
-    # Ignore types: https://github.com/PyCQA/pylint/issues/3167
     hass: HomeAssistant = None  # type: ignore
```

Comment:
The `None` assigned default + the ignore was part of the issue. If that is fixed, this can now be removed.

```suggestion
    hass: HomeAssistant
```

Reply:
I haven't checked if that's the case, but if it is, I'm very much fine with it. The "not purely typed" commentary in that case should likely be removed as well though. Feel free to commit and to this branch if you like :)

Reply:
Well removing the comment on a workaround without removing the workaround, is weird.

If you don't want to check or change it, I suggest you revert/remove those changes from this PR.

Reply:
The point of removing those comments is that they associate "Ignore types" i.e. IIUC `# type: ignore` with/as a workaround for pylint, and I don't think pylint has anything to do with those comments and it doesn't process them at all -- they're for mypy -- so therefore they're just misleading at best.

If I'm mistaken and pylint actually does something with `# type: ignore`, then I agree with you. Do you know how it is?

Reply:
The bug referenced here is:

False no-member positive on defaultless instance variables on subclasses


The None value is to workaround the linked issue by assigning a default value. The type ignore was to make CI this shut up about the workaround.

Reply:
Oh, I know the referenced bug, I reported it.

Seems we'll just have to disagree on the misattribution/misleadingness of the comment here, and it doesn't seem to me we're up to checking whether the None assignment could be removed in context of this PR, so I'll revert those changes (and address the CI failure) to get the other few bits in.

Reply:
🤷 Opened up #49334 instead

Reply:
LOL ok, looking at the wall of failures from the tests... makes me realize morning coffee ain't a bad thing... hahaha

Well, adjusting the above PR partly.